frontend/src/components/common/Notifications/index.tsx

```diff
@@ -0,0 +1,172 @@
+  clickEventHandler: (url?: string) => void
+) {
+  const { t } = useTranslation();
+  const classes = useStyles();
+  if (!notifications || notifications.length === 0) {
+    return (
+      <MenuItem onClick={() => clickEventHandler()} className={classes.notificationItem}>
+        {t(`frequent|You don't have any notifications right now`)}
+      </MenuItem>
+    );
+  }
+
+  return notifications?.map((notification, index: number) => {
+    return (
+      <MenuItem
+        onClick={() => clickEventHandler(notification.url)}
+        key={`${notification}__${index}`}
+        className={classes.notificationItem}
+      >
+        <Grid container justifyContent="flex-end" spacing={2}>
+          <Grid item md={12}>
+            {notification.message}
+          </Grid>
+          <Grid item>{timeAgo(notification.date)}</Grid>
+        </Grid>
+      </MenuItem>
+    );
+  });
+}
+
+export default function Notifications() {
+  const classes = useStyles();
+  const [anchorEl, setAnchorEl] = useState(null);
+  const history = useHistory();
+  const notifications = useTypedSelector(state => state.ui.notifications);
+  const [notificationAlert, setNotificationAlert] = useState(false);
+  const dispatch = useDispatch();
+  const [events, error] = Event.useList();
+
+  useEffect(() => {
+    if (events && events.length !== 0) {
+      const importantEvents: Notification[] = events
+        .filter((event: KubeEvent) => event.type !== 'Normal')
+        .sort(
+          (e1: KubeEvent, e2: KubeEvent) =>
+            new Date(e2.metadata.creationTimestamp).getTime() -
+            new Date(e1.metadata.creationTimestamp).getTime()
+        )
+        .map((event: KubeEvent) => {
+          const notification = new Notification();
+          notification.date = event.metadata.creationTimestamp;
+          notification.id = event.metadata.uid;
+          notification.message = event.message;
+          notification.url = '/';
+          return notification;
+        });
+      // we are here means the events list changed and we have now new set of events, so we will notify the store about it
+      dispatch(setUINotifications(importantEvents));
+    }
+  }, [events, error]);
+
+  useEffect(() => {
+    if (!notifications || notifications.length === 0) {
+      return;
+    }
+    // push notification only if we have some cluster selected + there are events
+    const recentNotificationFromStorage = localStorage.getItem('recent_notification');
+    if (recentNotificationFromStorage) {
+      const parsedRecentNotification = JSON.parse(recentNotificationFromStorage);
+      if (parsedRecentNotification?.id !== notifications[0].id) {
+        setNotificationAlert(true);
+      }
+    }
+    // we store the recent event in storage which we will use to check if there were new notifications loaded from last time the user checked the notification tab
+    localStorage.setItem('recent_notification', JSON.stringify(notifications[0]));
```

Comment:
Are the notification ordering stable and consistent or eventually consistent? If it's eventually consistent(logs, queues, third party systems), then I'm not entirely sure this logic will work out.

Reply:
When adding these notifications in redux store we always store them in order of the time they occured so the most recent ones stay on top of that list, so it eventually is consistent.But I am not sure if this is the problem you point to can you please elaborate more??

Reply:
This part isn't there anymore... resolving conversation.